package/gluon-mesh-batman-adv-core/src/respondd.c

```diff
@@ -212,11 +212,13 @@ static void add_gateway(struct json_object *obj) {
 
 	while (getline(&line, &len, f) >= 0) {
 		char addr[18];
+		char nexthop[18];
 
-		if (sscanf(line, "=> %17[0-9a-fA-F:]", addr) != 1)
+		if (sscanf(line, "=> %17[0-9a-fA-F:] (%*u) %17[0-9a-fA-F:]", addr, &metric, nexthop) != 3) {
```

Comment:
Another thing I'ce noticed: the metric might be less then 100, so it must be `( %*u)` instead of `(%*u)` to allow whitespace. Sorry, didn't see this before.